README.md

```diff
@@ -157,37 +189,160 @@ private AuthenticationCallback getAuthInteractiveCallback() {
+mSingleAccountApp.getCurrentAccountAsync(new ISingleAccountPublicClientApplication.CurrentAccountCallback() {
+    @Override
+    public void onAccountLoaded(@Nullable IAccount account) {
+        if (account != null) {
+            // Store the account for later use
+            mAccount = account;
+            // Account is signed in
+            String username = account.getUsername();
+        } else {
+            // No account is signed in
+        }
+    }
+
+    @Override
+    public void onAccountChanged(@Nullable IAccount priorAccount, @Nullable IAccount currentAccount) {
+        // Account has changed, update UI accordingly
+    }
+
+    @Override
+    public void onError(@NonNull MsalException exception) {
+        // Handle error loading account
+    }
+});
+```
+
+```java
+// Silent Token Acquisition
+AcquireTokenSilentParameters silentParameters = new AcquireTokenSilentParameters.Builder()
+    .withScopes(SCOPES)
+    .forAccount(account)
+    .withCallback(getAuthInteractiveCallback())
+    .build();
+mSingleAccountApp.acquireTokenSilentAsync(silentParameters);
+```
+
+```java
+// Silent Token Acquisition synchronously, must be done in a background thread.
+AcquireTokenSilentParameters silentParameters = new AcquireTokenSilentParameters.Builder()
+    .withScopes(SCOPES)
+    .forAccount(account)
+    .build();
+mSingleAccountApp.acquireTokenSilent(silentParameters);
+```
+
+All of the above examples can be found in the `snippets` directory, where we've also included jotlin examples.
+
+>**WARNING**: Device Code Flow (`acquireTokenWithDeviceCode`) is discouraged due to security concerns in the industry. Only use this authentication method in specific scenarios where devices lack input capabilities necessary for interactive authentication (e.g., conference room devices without text input). For standard Android applications, always use the interactive `acquireToken()` or `signIn()` APIs demonstrated above.
+
+>**IMPORTANT**: 
+>- Always use Parameters-based APIs instead of deprecated methods
+>- Validate PCA initialization before making any API calls
+>- Handle UI updates on the main thread using `activity.runOnUiThread`
+>- Refresh account lists after authentication operations
+>- Use proper callback interfaces for communication between components
+
+#### Configuration Best Practices
+
+1. **Authentication Configuration**:
+   - Enable broker integration for enhanced security and SSO capabilities
+   - URL encode special characters in `redirect_uri` within auth_config.json
+   - Do NOT URL encode the signature hash in AndroidManifest.xml
+
+2. **Resource Organization**:
+   - Use proper resource naming conventions (e.g., `activity_*`, `fragment_*`)
+   - Extract dimensions and strings to resource files
+   - Define consistent theme attributes
+   - Implement proper view binding
+
+3. **Error Handling**:
+   - Validate PCA initialization before API calls
+   - Handle and log authentication errors appropriately
```

Comment:
do we have any samples on this?

Reply:
not particularly, It's laregly up to how cline sets up the UI. I changed it to say "communicate" rather than "log" though